src/extension.ts

```diff
@@ -1309,20 +1309,133 @@ export async function activate(context: vscode.ExtensionContext) {
 	context.subscriptions.push(
 		vscode.commands.registerCommand(
 			'intuita.acceptFolder',
-			async (arg0, ...otherArgs) => {
-				const firstJobHash: string | null =
-					typeof arg0 === 'string' ? arg0 : null;
-				if (firstJobHash === null) {
-					throw new Error(
-						'Did not pass the jobHashes into the command.',
+			async (_arg0) => {
+				try {
+					const arg0: {
+						id: string;
+						caseHash: string;
+						jobHashes: JobHash[];
+					} | null = typeof _arg0 === 'object' ? _arg0 : null;
+					if (arg0 === null) {
+						throw new Error(
+							'Did not pass an object into the command.',
+						);
+					}
+					if (!arg0.id) {
+						throw new Error(
+							'Did not include id in the passed object.',
+						);
+					}
+					if (!arg0.caseHash) {
+						throw new Error(
+							'Did not include caseHash in the passed object.',
+						);
+					}
+					if (!arg0.jobHashes || arg0.jobHashes.length === 0) {
+						throw new Error(
+							'Did not include jobHashes in the passed object.',
+						);
+					}
+
+					const jobHashes = arg0.jobHashes.slice();
+
+					if (!repositoryService) {
+						throw new Error(
+							'Unable to initialize repositoryService',
+						);
+					}
+
+					const currentBranch = repositoryService.getCurrentBranch();
+
+					if (
+						currentBranch === null ||
+						currentBranch.name === undefined
+					) {
+						throw new Error('Unable to get current branch');
+					}
+
+					// @TODO for now stagedJobs = all case jobs
+					const stagedJobs = [];
+
+					for (const jobHash of jobHashes) {
+						const job = jobManager.getJob(jobHash);
+
+						if (job === null) {
+							continue;
+						}
+
+						stagedJobs.push({
+							hash: job.hash.toString(),
+							label: buildJobElementLabel(
+								job,
+								vscode.workspace.workspaceFolders?.[0]?.uri
```

Comment:
this should be calculate before the for-each loop

Reply:
Can you explain? Here we just iterate jobs and build labels for each job